annotations.md

```diff
@@ -0,0 +1,228 @@
+
+
+    Requirement : *Extract an entire configmap/Secret from the Kubernetes resource*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/dbConfiguration”: "path={.status.data.dbConfiguration},objectType=ConfigMap”
+    ```
+
+
+    Descriptor
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding
+    ```
+
+3. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as an environment variable
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as an environment variable.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},objectType=ConfigMap"
+    ```
+
+
+    Descriptor
+
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:certificate:bindAs=volume
+    ```
+
+
+4. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as a volume mount
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as a volume mount.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},bindAs=volume,objectType=ConfigMap"
+    ```
+
+
+    Descriptor
+
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:certificate:bindAs=volume
+    ```
+
+
+5. #### Use “db_timeout” from the ConfigMap “status.data.dbConfiguration” as “timeout” in the binding Secret.
+
+    Requirement: *Extract a specific field from the configmap/Secret from the Kubernetes resource and map it to different name in the binding Secret*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/timeout”: 
+    “path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=db_timeout”
+    ```
+
+
+    Descriptor
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:timeout:sourceKey=db_timeout
+    ```
+
+6. #### Use the attribute “status.data.url”
+
+    Requirement: *Extract a string from the Kubernetes resource.*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/url”:"path={.status.data.url}"
+    ```
+
+    Descriptor
+
+    ```
+    - path: data.uri
+      x-descriptors:
+        - servicebinding
+    ```
+
+7. #### Use the attribute “status.data.“connectionURL” as uri in the binding Secret
+
+    Requirement: *Extract a string from the Kubernetes resource, and map it to custom name in the binding Secret.*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/uri: "path={.status.data.connectionURL}”
+    ```
+
+
+
+    Descriptor
+
+    ```
+    - path: data.uri
+      x-descriptors:
+        - servicebinding:connectionURL
```

Comment:
perhaps this is reversed?  should be `data.connectionURL` and `servicebinding:uri`

Reply:
yup!